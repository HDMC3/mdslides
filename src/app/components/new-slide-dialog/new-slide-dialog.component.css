nb-card {
    min-width: 90vw;
}

form {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
}

.dialog-footer {
    display: flex;
    justify-content: space-between;
}

@media screen and (min-width: 500px) {
    nb-card {
        min-width: 350px;
    }
}