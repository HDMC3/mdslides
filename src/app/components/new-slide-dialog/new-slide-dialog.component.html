<nb-card>
    <nb-card-header>
        <h6>
            Nueva diapositiva
        </h6>
    </nb-card-header>
    <nb-card-body>
        <form>
            <label for="name">Nombre:</label>
            <input [formControl]="nameControl" id="name" nbInput type="text" [status]="nameControlStatus" autocomplete="off" (keyup.enter)="save()">
            <small *ngIf="!nameControl.pristine && nameControl.invalid" style="color: var(--color-danger-500)">El nombre es requerido</small>
        </form>
    </nb-card-body>
    <nb-card-footer>
        <div class="dialog-footer">
            <button nbButton status="danger" (click)="closeDialog()">Cancelar</button>
            <button nbButton status="success" (click)="save()">Guardar</button>
        </div>
    </nb-card-footer>
</nb-card>