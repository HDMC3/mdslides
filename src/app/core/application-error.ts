export class ApplicationError extends Error {
    constructor(message: string, public title?: string) {
        super(message);
    }
}
