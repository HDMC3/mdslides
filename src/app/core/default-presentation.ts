import { v4 } from 'uuid';
import { Presentation } from '../data/interfaces/presentation';

export const getDefaultPresentation = (id: string): Presentation => {
    return {
        id,
        title: 'Titulo de presentacion',
        creation_date: new Date(),
        modification_date: new Date(),
        description: '',
        slides: [
            {
                id: v4(),
                name: 'Presentacion inicial',
                code: [
                    '---',
                    'style: |',
                    'section {',
                    '   display: flex;',
                    '   flex-direction: column;',
                    '   align-items: center;',
                    '   justify-content: center;',
                    '}',
                    '---',
                    '',
                    '# Inicio de la presentacion',
                    '# ðŸ˜„âœŒ'
                ]
            }
        ]
    };
};
