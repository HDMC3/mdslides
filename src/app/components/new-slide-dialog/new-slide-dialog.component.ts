import { Component } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { NbComponentStatus, NbDialogRef } from '@nebular/theme';
import { Subscription } from 'rxjs';
import { FormValidators } from 'src/app/shared/form-validators';

@Component({
    selector: 'app-new-slide-dialog',
    templateUrl: './new-slide-dialog.component.html',
    styleUrls: ['./new-slide-dialog.component.css']
})
export class NewSlideDialogComponent {

    nameControlStatus: NbComponentStatus;
    nameControl: FormControl;
    nameControlSubscription: Subscription;

    constructor(
        private nbDialogRef: NbDialogRef<any>
    ) {
        this.nameControlStatus = 'basic';
        this.nameControl = new FormControl('', [Validators.required, FormValidators.noEmpty]);

        this.nameControlSubscription = this.nameControl.statusChanges.subscribe(status => {
            this.nameControlStatus = status === 'INVALID' ? 'danger' : 'basic';
        });
    }

    closeDialog() {
        this.nbDialogRef.close();
    }

    save() {
        if (this.nameControl.invalid) {
            this.nameControl.markAsPristine();
            this.nameControlStatus = 'danger';
            return;
        };
        this.nbDialogRef.close({
            name: this.nameControl.value.trim()
        });
    }

}
