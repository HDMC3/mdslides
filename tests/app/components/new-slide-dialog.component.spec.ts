import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NbButtonModule, NbCardModule, NbDialogRef, NbInputModule, NbThemeModule } from '@nebular/theme';
import { Observable } from 'rxjs';

import { NewSlideDialogComponent } from '../../../src/app/components/new-slide-dialog/new-slide-dialog.component';

describe('NewSlideDialogComponent', () => {
    let component: NewSlideDialogComponent;
    let fixture: ComponentFixture<NewSlideDialogComponent>;
    let compiled: HTMLElement;
    let componentNbDialogRef: NbDialogRef<any>;

    const nbDialogRef = {
        close() {},
        componentRef: {},
        onBackdropClick: new Observable<MouseEvent>(),
        onClose: new Observable<any>()
    };

    beforeEach(async() => {
        await TestBed.configureTestingModule({
            declarations: [NewSlideDialogComponent],
            imports: [
                FormsModule,
                ReactiveFormsModule,
                NbThemeModule.forRoot({ name: 'default' }),
                NbCardModule,
                NbInputModule,
                NbButtonModule
            ],
            providers: [
                { provide: NbDialogRef, useValue: nbDialogRef }
            ]
        })
            .compileComponents();

        componentNbDialogRef = TestBed.inject(NbDialogRef);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(NewSlideDialogComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
        compiled = fixture.nativeElement;
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(compiled).toMatchSnapshot();
    });

    it('should not close modal when form is invalid', () => {
        jest.spyOn(componentNbDialogRef, 'close');
        const buttons = compiled.querySelectorAll<HTMLButtonElement>('.dialog-footer button');
        const saveButton = buttons[1];
        component.nameControl.setValue('');
        saveButton.click();
        expect(componentNbDialogRef.close).not.toBeCalled();
    });

    it('should close modal with slide name value when form is valid', () => {
        jest.spyOn(componentNbDialogRef, 'close');
        const nameNewSlide = 'Nombre de diapositiva';
        const buttons = compiled.querySelectorAll<HTMLButtonElement>('.dialog-footer button');
        const saveButton = buttons[1];
        component.nameControl.setValue(nameNewSlide);
        saveButton.click();
        expect(componentNbDialogRef.close).toBeCalledWith({ name: nameNewSlide });
    });
});
