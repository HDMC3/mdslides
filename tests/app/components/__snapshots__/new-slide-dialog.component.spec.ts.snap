// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSlideDialogComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <nb-card>
    <nb-card-header>
      <h6>
         Nueva diapositiva 
      </h6>
    </nb-card-header>
    <nb-card-body>
      <form
        class="ng-untouched ng-pristine ng-valid"
        novalidate=""
      >
        <label
          for="name"
        >
          Nombre:
        </label>
        <input
          autocomplete="off"
          class="ng-untouched ng-pristine ng-invalid size-medium status-basic shape-rectangle"
          id="name"
          nbinput=""
          type="text"
        />
        
      </form>
    </nb-card-body>
    <nb-card-footer>
      <div
        class="dialog-footer"
      >
        <button
          aria-disabled="false"
          class="appearance-filled size-medium shape-rectangle status-danger"
          nbbutton=""
          status="danger"
          tabindex="0"
        >
          Cancelar
        </button>
        <button
          aria-disabled="false"
          class="status-success appearance-filled size-medium shape-rectangle"
          nbbutton=""
          status="success"
          tabindex="0"
        >
          Guardar
        </button>
      </div>
    </nb-card-footer>
  </nb-card>
</div>
`;
